### PR TITLE
Initial WebSocket Implementation

### PR DESCRIPTION
**Pull request by JBassett**
## Summary

Adding initial support for the Web Socket API.  For now the only service calls that have been moved over are getting the config, get services, and getting all states.  A future PR will add support for subscribing to events via the websocket.

## Screenshots

N/A

## Link to pull request in Documentation repository

N/A

### CODE REVIEW

File: common/src/main/java/io/homeassistant/companion/android/common/data/HomeAssistantApis.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        var cookieManager: CookieManager? = null
+        try {
+            cookieManager = CookieManager.getInstance()
+        } catch (e: Exception) {
+            // Noop
+        }
+        if (cookieManager != null) {
+            builder.cookieJar(CookieJarCookieManagerShim())
+        }
+        builder.callTimeout(30L, TimeUnit.SECONDS)
```
Comment (nick2525) [inline common/src/main/java/io/homeassistant/companion/android/common/data/HomeAssistantApis.kt:65]: Better to put in constants in this file

File: common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+import kotlinx.coroutines.withTimeout
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.WebSocket
+import okhttp3.WebSocketListener
+import okio.ByteString
+import javax.inject.Inject
+
+class WebSocketRepositoryImpl @Inject constructor(
```
Comment (nick2525) [inline common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt:30]: maybe put WebSocket handle in separate gradle module?

Comment (JBassett) [reply common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt:33]: Once we get the library totally flushed out internally I can see us extracting it into it's own repo but for now I'm just as content leaving it tightly coupled.  As we add more we are going to run into issues that are just easier to tackle while it's internal.

File: common/src/main/java/io/homeassistant/companion/android/common/data/HomeAssistantApis.kt (hunk 1 above)
Comment (JBassett) [reply common/src/main/java/io/homeassistant/companion/android/common/data/HomeAssistantApis.kt]: Done

**Review by dshokouhi — APPROVED** [review]
Ran a quick test on my phone and didn't seem to see an issue here in any place that we use `getServices` and `getEntities` all seems to be working well :) Great job!!!

**Pull request merged**